polar-language-server/src/lib.rs

```diff
@@ -141,61 +165,81 @@ impl PolarLanguageServer {
             // Nothing to do when we receive the `Initialized` notification.
             Initialized::METHOD => (),
 
-            _ => log(&format!("on_notification {} {:?}", method, params)),
+            _ => log("unexpected notification"),
         }
     }
 }
 
 /// Individual LSP notification handlers.
 impl PolarLanguageServer {
     fn on_did_open_text_document(&mut self, doc: TextDocumentItem) -> Diagnostics {
-        if let Some(TextDocumentItem { uri, .. }) = self.upsert_document(doc) {
-            log(&format!("reopened tracked doc: {}", uri));
+        log(&format!("\topening: {}", doc.uri));
+        if self.upsert_document(doc).is_some() {
+            log("\t\treopened tracked doc");
         }
         self.reload_kb()
     }
 
     fn on_did_change_text_document(&mut self, doc: TextDocumentItem) -> Diagnostics {
         let uri = doc.uri.clone();
         if self.upsert_document(doc).is_none() {
-            log(&format!("updated untracked doc: {}", uri));
+            log(&format!("\tupdated untracked doc: {}", uri));
         }
         self.reload_kb()
     }
 
-    fn on_did_delete_files(&mut self, uris: Vec<Url>) -> Diagnostics {
+    fn on_did_change_watched_files(&mut self, uris: Vec<Url>) -> Diagnostics {
```

Comment:
Could you add a comment here saying that we currently only listen for deletions, so this method assumes all changes are deletions?

Reply:
Sure, I can link back to this comment: https://github.com/osohq/oso/blob/9c0400beecd94697e36923e5f29d78fc4ca0ae86/polar-language-server/src/lib.rs#L95-L97

Reply:
Yes please. It took me a little while to double-check the logic here and that this wasn't called elsewhere

Reply:
3267aca221f90c09afd8c7a114d57cb506f0e3cc